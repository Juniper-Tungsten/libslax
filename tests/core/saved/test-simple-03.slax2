version 1.1;

match doc {
    <top> {
        if (this && that || the/other[one]) {
            /* SLAX has a simple "if" statement */
            <simple> "if";
        
        } else if (yet[more == "fun"]) {
            /* ... and it has "else if" */
        
        } else {
            /* ... and "else" */
        }
        if (starts-with(name, "fe-")) {
            if (mtu < 1500) {
                /* Deal with fast ethernet interfaces with low MTUs */
            }
        
        } else {
            if (mtu > 8096) {
                /* Deal with non-fe interfaces with high MTUs */
                <mtu> mtu;
            }
        }
    }
}
