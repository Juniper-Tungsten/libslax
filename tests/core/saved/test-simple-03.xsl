<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
  <xsl:template match="doc">
    <top>
      <xsl:choose>
        <xsl:when test="this and that or the/other[one]">
          <!-- SLAX has a simple "if" statement -->
          <simple>if</simple>
        </xsl:when>
        <xsl:when test="yet[more = &quot;fun&quot;]">
          <!-- ... and it has "else if" -->
        </xsl:when>
        <xsl:otherwise>
          <!-- ... and "else" -->
        </xsl:otherwise>
      </xsl:choose>
      <xsl:choose>
        <xsl:when test="starts-with(name, &quot;fe-&quot;)">
          <xsl:if test="mtu &lt; 1500">
            <!-- Deal with fast ethernet interfaces with low MTUs -->
          </xsl:if>
        </xsl:when>
        <xsl:otherwise>
          <xsl:if test="mtu &gt; 8096">
            <!-- Deal with non-fe interfaces with high MTUs -->
            <mtu>
              <xsl:value-of select="mtu"/>
            </mtu>
          </xsl:if>
        </xsl:otherwise>
      </xsl:choose>
    </top>
  </xsl:template>
</xsl:stylesheet>
