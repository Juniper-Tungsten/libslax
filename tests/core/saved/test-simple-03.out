<?xml version="1.0"?>
<top>
  <simple>if</simple>
  <mtu>10000</mtu>
</top>
